File: qc-core/src/main/java/com/eager/questioncloud/core/domain/user/infrastructure/repository/UserJpaRepository.java
package com.eager.questioncloud.core.domain.user.infrastructure.repository;

import com.eager.questioncloud.core.domain.user.enums.AccountType;
import com.eager.questioncloud.core.domain.user.infrastructure.entity.UserEntity;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface UserJpaRepository extends JpaRepository<UserEntity, Long> {
    Boolean existsByUserAccountInformationEntityAccountTypeAndUserAccountInformationEntitySocialUid(AccountType accountType, String socialUid);

    Boolean existsByUserInformationEntityPhone(String phone);

    Boolean existsByUserInformationEntityEmail(String email);

    Optional<UserEntity> findByUserAccountInformationEntityAccountTypeAndUserAccountInformationEntitySocialUid(AccountType accountType,
        String socialUid);

    Optional<UserEntity> findByUserInformationEntityEmail(String email);

    Optional<UserEntity> findByUserInformationEntityPhone(String phone);
}

File: qc-core/src/main/kotlin/com/eager/questioncloud/core/domain/user/infrastructure/repository/UserJpaRepository.kt
package com.eager.questioncloud.core.domain.user.infrastructure.repository

import com.eager.questioncloud.core.domain.user.enums.AccountType
import com.eager.questioncloud.core.domain.user.infrastructure.entity.UserEntity
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface UserJpaRepository : JpaRepository<UserEntity, Long> {
    fun existsByUserInformationEntityPhone(phone: String): Boolean

    fun existsByUserInformationEntityEmail(email: String): Boolean

    fun findByUserAccountInformationEntityAccountTypeAndUserAccountInformationEntitySocialUid(
        accountType: AccountType,
        socialUid: String
    ): UserEntity?

    fun findByUserInformationEntityEmail(email: String): Optional<UserEntity>

    fun findByUserInformationEntityPhone(phone: String): Optional<UserEntity>

    fun findByUidIn(ids: List<Long>): List<UserEntity>
}
